pallets/connectors/src/lib.rs

```diff
@@ -0,0 +1,507 @@
+				},
+				domain,
+			)?;
+
+			Ok(())
+		}
+
+		/// Update a token price
+		#[pallet::weight(< T as Config >::WeightInfo::update_token_price())]
+		pub fn update_token_price(
+			origin: OriginFor<T>,
+			pool_id: PoolIdOf<T>,
+			tranche_id: TrancheIdOf<T>,
+			domain: Domain,
+		) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+
+			ensure!(
+				T::PoolInspect::tranche_exists(pool_id, tranche_id),
+				Error::<T>::TrancheNotFound
+			);
+
+			let latest_price = T::PoolInspect::get_tranche_token_price(pool_id, tranche_id)
+				.ok_or(Error::<T>::MissingPrice)?;
+
+			Self::do_send_message(
+				who,
+				Message::UpdateTokenPrice {
+					pool_id,
+					tranche_id,
+					price: latest_price.price,
+				},
+				domain,
+			)?;
+
+			Ok(())
+		}
+
+		/// Update a member
+		#[pallet::weight(< T as Config >::WeightInfo::update_member())]
+		pub fn update_member(
+			origin: OriginFor<T>,
+			address: DomainAddress<Domain>,
+			pool_id: PoolIdOf<T>,
+			tranche_id: TrancheIdOf<T>,
+			valid_until: Moment,
+		) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+
+			// Check that the origin is a member of this tranche token or
+			// is a MemberListAdmin and thus allowed to add other members.
+			ensure!(
+				T::Permission::has(
+					PermissionScope::Pool(pool_id),
+					who.clone(),
+					Role::PoolRole(PoolRole::TrancheInvestor(tranche_id, Self::now()))
+				) || T::Permission::has(
+					PermissionScope::Pool(pool_id),
+					who.clone(),
+					Role::PoolRole(PoolRole::MemberListAdmin)
+				),
+				BadOrigin
+			);
+
+			T::Permission::add(
+				PermissionScope::Pool(pool_id),
+				address.into_account_truncating(),
+				Role::PoolRole(PoolRole::TrancheInvestor(tranche_id, valid_until)),
+			)?;
+
+			Self::do_send_message(
+				who,
+				Message::UpdateMember {
+					pool_id,
+					tranche_id,
+					valid_until,
+					address: address.address,
+				},
+				address.domain,
+			)?;
+
+			Ok(())
+		}
+
+		/// Transfer tranche tokens to a given address
+		#[pallet::weight(< T as Config >::WeightInfo::transfer())]
+		pub fn transfer(
+			origin: OriginFor<T>,
+			pool_id: PoolIdOf<T>,
+			tranche_id: TrancheIdOf<T>,
+			address: DomainAddress<Domain>,
+			amount: <T as pallet::Config>::Balance,
+		) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+
+			// Check that the destination is a member of this tranche token
+			ensure!(
+				T::Permission::has(
+					PermissionScope::Pool(pool_id),
+					address.into_account_truncating(),
+					Role::PoolRole(PoolRole::TrancheInvestor(tranche_id, Self::now()))
+				),
+				BadOrigin
+			);
+
+			ensure!(!amount.is_zero(), Error::<T>::InvalidTransferAmount);
+
+			// TODO: Transfer to the domain account for bookkeeping
+			// T::Tokens::transfer(
+			// 	T::CurrencyId::Tranche(pool_id, tranche_id),
+			// 	&who,
+			// 	&DomainLocator {
+			// 		domain: address.domain,
+			// 	}
+			// 	.into_account_truncating(),
+			// 	amount,
+			// 	false,
+			// )?;
```

Comment:
I am working on the currencyId enum anyways this week. We could make the `TrancheCurrency` a struct which connectors directly depends on and it will implement `Into<CurrencyId>`. 

Other solution, we extend the `PoolInspect` trait and let it return `TrancheCurrency` and `PoolCurrency` for a pool, if it exists. Then you could simply use the returned `Currency`.

Reply:
Ah yes that would be nice @mustermeiszer 😃

Reply:
Which one do you prefer? I am still in the process of making up my mind around the changes to `CurrencyId`. 

I have two options in my head: 

### First - structured currency types that are wrapped in enums
```rust
// Also implements `Into<CurrencyId`
struct TrancheCurrency<PoolId, TrancheId> {
   pool_id: PoolId, 
   tranche_id: TrancheId
}

impl<PoolId, TrancheId> TrancheCurrency<PoolId, TrancheId> {
   const DECIMALS: u128 = ???; // Need to get this from the registry
   const SYMBOL: &str =  ???; //Need to get this from the registry
}

// Also implements `Into<CurrencyId`
struct KSM;

impl KSM {
   const DECIMALS: u128 = primitives::currency_decimals::KSM;
   const SYMBOL: &str =  &"KSM";
}

// Also implements `Into<CurrencyId`
struct AUSD;

impl AUSD {
   const DECIMALS: u128 = primitives::currency_decimals::AUSD;
   const SYMBOL: &str =  &"AUSD";
}

// Currencies that can. beused in a pool as a base currency
pub enum PoolCurrency {
   KSM(KSM),
   AUSD(AUSD)
}

// The currencies one can invest into 
pub enum InvestmentId {
   Tranche(TrancheCurrency<PoolId, TrancheId>)
}

// The overarching CurrencyId enum that is only used by pallets
// that need knowledge of all currencies, e.g. orml_tokens
//
// MUST NOT be differentiate between runtimes, but could be
mod altair {
   enum CurrencyId {
     KSM(KSM),
     AUSD(AUSD),
     TrancheCurrency(TrancheCurrency)
   }
}
```

### Second - Do not use structured currencies but rather have the enums convert it

Reply:
@mustermeiszer do you mind creating an issue to discuss this? 🙏

Reply:
Cool, thanks! @mustermeiszer I might ping you on this one :)

Reply:
@NunoAlexandre do we still want to address this TODO here? I think it should be possible now that https://github.com/centrifuge/centrifuge-chain/pull/995 was merged